### PR TITLE
docs: argo workflow and events docs

### PR DESCRIPTION
**Pull request by andrrax**
Initial workflow and events docs, as well as some pre-commit linting.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -11,3 +11,17 @@ repos:
     hooks:
       - id: yamllint
         args: [-c, .yamllint.yaml]
+  - repo: https://github.com/PyCQA/flake8
+    rev: 7.0.0
+    hooks:
+      - id: flake8
+        args:
+          - --max-line-length=120
```
Comment (cardoe) [inline .pre-commit-config.yaml:19]: Probably should put this in a config file so we can have an editor pick it up to help folks do the right thing.

**Review by cardoe — COMMENTED** [review]
Just some notes

File: argo-workflows/generic/workflowtemplates/get-obm-creds.yaml [hunk 2]
```diff
@@ -19,8 +23,36 @@ spec:
...
+              f.write(secret_name)
+        env:
+        # pass this Pod's uid into the pod env to make it available to the argo secrets method, otherwise pod get
+        # permissions are required for the service account specified above.
+        - name: KUBERNETES_POD_UID
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.uid
+        # only needed in this example Template as it's written using a ScriptTemplate which apparently doesn't seem
+        # to honor the workingDir directive, and mounts the above script at /argo/staging/script.
```
[the remark this answers is not in the document]
Comment (andrrax) [reply argo-workflows/generic/workflowtemplates/get-obm-creds.yaml:56]: which part, the metadata.uid, or the comment about the ScriptTemplate?

[the remark this answers is not in the document]
Comment (cardoe) [reply argo-workflows/generic/workflowtemplates/get-obm-creds.yaml:56]: The ScriptTemplate path. The `metadata.uid` is using the Downward API and that makes sense.

**Review by cardoe — APPROVED** [review]

**Pull request merged**